Assets/_Scripts/Character/StatusAilment.cs

```diff
@@ -0,0 +1,19 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public abstract class StatusAilment {
+	public string name;
+	public int duration;
+	public int level;
+	public BattlePhase phase;
```

Comment:
Instead of holding a phase, I think it would make more sense to create your own enum for status ailment type. What if there is a status ailment that we don't want to decrement from DecrementAllAilmentsDuration() ? What if a status ailment is a turn-modifying status ailment (e.g. sleep/confuse). We will have a lot more freedom if we do this.

Reply:
TurnController needs someway to represent the current stage of the turn. I can create my own enum, but it's exactly based on each step of the turn, not specifically for status ailments.

As for freedom of status ailments, you can override the decrement functions to resolve any special attributes.

For turn-modifying status ailments, that's exactly why I'm using BattlePhase. The status ailments are ran during that specific stage - should disable/modify any UIs related to that stage.